aspnetcore/fundamentals/servers/kestrel.md

```diff
@@ -0,0 +1,139 @@
+---
+title: Kestrel server
+author: tdykstra
+ms.author: tdykstra
+manager: wpickett
+ms.date: 10/27/2016
+ms.topic: article
+ms.assetid: 560bd66f-7dd0-4e68-b5fb-f31477e4b785
+ms.prod: aspnet-core
+uid: fundamentals/servers/kestrel
+---
+# Kestrel server for ASP.NET Core
+
+By [Tom Dykstra](http://github.com/tdykstra), [Chris Ross](https://github.com/Tratcher), and [Stephen Halter](https://twitter.com/halter73)
+
+Kestrel is a cross-platform [web server for ASP.NET Core](overview.md) based on [libuv](https://github.com/libuv/libuv), a cross-platform asynchronous I/O library. Kestrel is the web server that is included by default in ASP.NET Core new-project templates. 
+
+Kestrel supports the following features:
+
+  * HTTPS
+  * Opaque upgrade used to enable [WebSockets](https://github.com/aspnet/websockets)
+  * Unix sockets for high performance behind Nginx 
+
+Kestrel is supported on all platforms and versions that .NET Core supports.
+
+[View or download sample code](https://github.com/aspnet/Docs/tree/master/aspnetcore/fundamentals/servers/kestrel/sample)
+
+## When to use Kestrel with a reverse proxy
+
+If your application accepts requests only from an internal network, you can use Kestrel by itself.
+
+![Kestrel to internal network](kestrel/_static/kestrel-to-internal.png)
+
+If you expose your application to the Internet, we recommend that you use IIS, Nginx, or Apache as a *reverse proxy server*. A reverse proxy server receives HTTP requests from the Internet and forwards them to Kestrel after some preliminary handling.
+
+![Kestrel to Internet](kestrel/_static/kestrel-to-internet.png)
+
+The most important reason for using a reverse proxy for edge deployments (exposed to traffic from the Internet) is security. Kestrel is relatively new and does not yet have a full complement of defenses against attacks. This includes but isn't limited to appropriate timeouts, size limits, and concurrent connection limits.
+
+Another scenario that requires a reverse proxy is when you have multiple applications that share the same port running on a single server. That doesn't work with Kestrel directly because Kestrel doesn't support host-header routing. When you configure Kestrel to listen on a port, it handles all traffic for that port regardless of host header.
```

Comment:
That doesn't work with Kestrel directly because Kestrel doesn't support sharing a port between multiple processes. A reverse proxy that can share ports must then forward to Kestrel on a unique port.